### ISSUE TITLE
lint/correctness/noUndeclaredVariables should ignore TsGlobalDeclaration

### ISSUE THREAD

**Issue by ash0080** [biomejs/biome#2932]
https://biomejs.dev/playground/?lineWidth=120&indentStyle=space&quoteStyle=single&semicolons=as-needed&attributePosition=multiline&bracketSameLine=true&lintRules=all&files.data.tsx=ZABlAGMAbABhAHIAZQAgAGcAbABvAGIAYQBsACAAewAKACAAIAAgACAAaQBuAHQAZQByAGYAYQBjAGUAIABGAGwAbwB3AEMAbwBuAGYAaQBnACAAewAKACAAIAAgACAAIAAgACAAIABmAGwAbwB3AFAAcgBvAHAAcwA6ACAAcwB0AHIAaQBuAGcACgAgACAAIAAgAH0ACgB9AA%3D%3D

Regarding similar situations, should the same requirement of "no undeclared variables" also be satisfied?

_Originally posted by @ash0080 in https://github.com/biomejs/biome/issues/1651#issuecomment-2121831887_

**Comment by Conaclos** [comment]
So you mean that `noUndeclaredVariables` should ignore ambient globals declared in a global declaration file?

**Comment by Conaclos** [comment]
Please update the playground link, it seems not illustrate the current issue.

**Comment by ash0080** [comment]
> Please update the playground link, it seems not illustrate the current issue.

It seems the playground server was experiencing an issue, but it appears to be accessible now.

**Comment by Conaclos** [comment]
I see the following code in the playground:

```ts
export default {
  flowProps: 'a'
} satisfies FlowConfig
```

I don't understand how this code is related to the current issue.

**Comment by ash0080** [comment]
> I see the following code in the playground:
> 
> ```ts
> export default {
>   flowProps: 'a'
> } satisfies FlowConfig
> ```
> 
> I don't understand how this code is related to the current issue.

check the main.ts
```ts
declare global {
    interface FlowConfig {
        flowProps: string
    }
}
```
In this case, where I have defined a global type instead of a global variable, the current implementation of Biome does not support this. The key difference from the previous example (#1651) is that I am now working with a **global type definition**, rather than a **global variable**.

**Comment by ematipico** [comment]
I can't replicate the issue in the playground.

**Comment by ash0080** [comment]
> I can't replicate the issue in the playground.

Try this one plz

https://biomejs.dev/playground/?lineWidth=120&indentStyle=space&quoteStyle=single&semicolons=as-needed&attributePosition=multiline&bracketSameLine=true&lintRules=all&files.data.tsx=ZQB4AHAAbwByAHQAIABkAGUAZgBhAHUAbAB0ACAAewAKACAAIABmAGwAbwB3AFAAcgBvAHAAcwA6ACAAJwBhACcACgB9ACAAcwBhAHQAaQBzAGYAaQBlAHMAIABGAGwAbwB3AEMAbwBuAGYAaQBnAA%3D%3D&files.main.tsx=ZABlAGMAbABhAHIAZQAgAGcAbABvAGIAYQBsACAAewAKACAAIAAgACAAaQBuAHQAZQByAGYAYQBjAGUAIABGAGwAbwB3AEMAbwBuAGYAaQBnACAAewAKACAAIAAgACAAIAAgACAAIABmAGwAbwB3AFAAcgBvAHAAcwA6ACAAcwB0AHIAaQBuAGcACgAgACAAIAAgAH0ACgB9AA%3D%3D

**Comment by ematipico** [comment]
@ash0080 sorry, the playground doesn't show the error.

Please create a repository at this point.
